src/Kernel32.Tests/IgnoreOnOsVersionUnderFactAttribute.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+using System;
+using Xunit;
+
+/// <summary>
+///     A variant of XUnit <see cref="FactAttribute" /> that is automatically skiped when the OS version is under a
+///     specific treshold.
+/// </summary>
+internal class IgnoreOnOsVersionUnderFactAttribute : FactAttribute
+{
+    private static readonly Version OsVersion = Environment.OSVersion.Version;
+    private readonly Version minimumVersion;
+
+    /// <summary>Initializes a new instance of the <see cref="IgnoreOnOsVersionUnderFactAttribute" /> class.</summary>
+    /// <param name="minimumVersion">
+    ///     The minimum version required for this <see cref="FactAttribute" /> in a format usable by
+    ///     <see cref="Version.Parse" />.
+    /// </param>
+    public IgnoreOnOsVersionUnderFactAttribute(string minimumVersion)
+    {
+        this.minimumVersion = Version.Parse(minimumVersion);
+    }
+
+    /// <inheritdoc />
+    public override string Skip
```

Comment:
I'll volunteer that I've [done something similar](https://github.com/AArnott/PCLCrypto/blob/master/src/PCLCrypto.Tests.Shared/ECDiffieHellmanTests.cs#L25), but without a custom attribute. 
It wouldn't work here because what you need is a runtime check. But I thought I'd share.